src/snyk/common/languageServer/languageServer.ts

```diff
@@ -72,6 +64,12 @@ export class LanguageServer implements ILanguageServer {
     // Create the language client and start the client.
     this.client = this.languageClientAdapter.create('Snyk LS', SNYK_LANGUAGE_SERVER_NAME, serverOptions, clientOptions);
 
+    this.client.onNotification('hasAuthenticated', ({ token }: { token: string }) => {
+      this.authenticationService.updateToken(token).catch(error => {
+        throw error;
```

Comment:
You may want to consider using `ErrorHandler`

Reply:
That's handy!